bdai_ros2_wrappers/bdai_ros2_wrappers/tf_listener_wrapper.py

```diff
@@ -1,116 +1,60 @@
 # Copyright (c) 2023 Boston Dynamics AI Institute Inc.  All rights reserved.
-import time
-from typing import Optional, Tuple
 
-import rclpy
+from typing import Optional
+
 from geometry_msgs.msg import TransformStamped
 from rclpy.node import Node
+from rclpy.task import Future
 from rclpy.time import Duration, Time
-from tf2_ros import ExtrapolationException, TransformException
 from tf2_ros.buffer import Buffer
 from tf2_ros.transform_listener import TransformListener
 
-import bdai_ros2_wrappers.process as process
+import bdai_ros2_wrappers.scope as scope
+from bdai_ros2_wrappers.futures import wait_for_future
 
 
-class TFListenerWrapper(object):
-    def __init__(
-        self,
-        node: Optional[Node] = None,
-        wait_for_transform: Optional[Tuple[str, str]] = None,
-        cache_time_s: Optional[int] = None,
-    ) -> None:
-        node = node or process.node()
+class TFListenerWrapper:
+    def __init__(self, node: Optional[Node] = None, cache_time_s: Optional[float] = None) -> None:
+        node = node or scope.node()
         if node is None:
-            raise ValueError("No process-wide ROS 2 node available (did you use bdai_ros2_wrapper.process.main?)")
+            raise ValueError("no ROS 2 node available (did you use bdai_ros2_wrapper.process.main?)")
         self._node = node
-
+        cache_time_py = None
         if cache_time_s is not None:
-            cache_time_py = Duration(seconds=cache_time_s)
-        else:
-            cache_time_py = None
+            cache_time_py = Duration(nanoseconds=cache_time_s * 1e9)
         self._tf_buffer = Buffer(cache_time=cache_time_py)
-        self._tf_listener = TransformListener(self._tf_buffer, self._node, spin_thread=False)
-
-        if wait_for_transform is not None and len(wait_for_transform) == 2:
-            self.wait_for_init(wait_for_transform[0], wait_for_transform[1])
+        self._tf_listener = TransformListener(self._tf_buffer, self._node)
```

Comment:
Is this being spun by its internal spin thread or the process-wide machinery?

Reply:
A few lines above this you can see that the current scope main node is fetched if no node is explicitly provided to the wrapper. The listener is bound to either. It doesn't spin up any threads, it relies on the executor the node is (or will be) associated with.

Reply:
For some reason, I thought that the TransformListener defaulted to the spin_thread being `True`.